crates/aptos-rosetta/src/types/misc.rs

```diff
@@ -315,3 +326,80 @@ pub async fn get_stake_balances(
         Ok(None)
     }
 }
+
+pub async fn get_delegation_stake_balances(
+    rest_client: &aptos_rest_client::Client,
+    account_identifier: &AccountIdentifier,
+    owner_address: AccountAddress,
+    pool_address: AccountAddress,
+    version: u64,
+) -> ApiResult<Option<BalanceResult>> {
+    let mut lockup_expiration: u64 = 0;
+    let mut requested_balance: Option<String> = None;
+
+    // get requested_balance
+    if let Ok(balances_response) = rest_client.view(
+        &ViewRequest {
+            function: DELEGATION_POOL_GET_STAKE_FUNCTION.clone(),
+            type_arguments: vec![],
+            arguments: vec![
+                serde_json::Value::String(String::from(
+                    pool_address.to_string(),
+                )),
+                serde_json::Value::String(String::from(
+                    owner_address.to_string()
+                )),
+            ],
+        },
+        Some(version),
+    ).await {
+        let result = balances_response.into_inner();
+
+        if account_identifier.is_delegator_active_stake() {
+            requested_balance = result.get(0).map(|v| v.to_string());
+        } else if account_identifier.is_delegator_inactive_stake() {
+            requested_balance = result.get(1).map(|v| v.to_string());
+        } else if account_identifier.is_delegator_pending_inactive_stake() {
+            requested_balance = result.get(2).map(|v| v.to_string());
+        }
+    } else {
+        return Err(ApiError::InternalError(Some(
+            "Unable to get delegation stake balances".to_string(),
+        )));
+    }
```

Comment:
This still discards the error from `rest_client.view`. You can try `.await?`, which throws the error if there is one.:
```suggestion
    let balances_response = rest_client.view(
        &ViewRequest {
            function: DELEGATION_POOL_GET_STAKE_FUNCTION.clone(),
            type_arguments: vec![],
            arguments: vec![
                serde_json::Value::String(String::from(
                    pool_address.to_string(),
                )),
                serde_json::Value::String(String::from(
                    owner_address.to_string()
                )),
            ],
        },
        Some(version),
    ).await?;
    let result = balances_response.into_inner();
    ...
```